src/darwin/Framework/CHIP/CHIPDevice.h

```diff
@@ -78,16 +75,49 @@ extern NSString * const kCHIPStatusKey;
 /**
  * Read attribute in a designated attribute path
  *
- * @param completion  response handler will receive either value or error. value will be an NSArray object with NSDictionary
- * elements. Each NSDictionary will have "endpointId", "clusterId", "attributeId", "status" and "data" keys. "endpointId",
- * "clusterId", "attributeId" and "status" will be mapped to NSNumber objects. "status" with 0 value indicates success and non-zero
- * value indicates failure. "data" key is present only when "status" value is 0. "data" key will be mapped to an NSDictionary
- * object, representing attribute value of the path. NSDictionary representing attribute value will contain "type" and "value" keys.
- *                        "type" will be mapped to "SignedInteger", "UnsignedInteger", "UTF8String", "OctetString", "Float",
- * "Double", "Boolean", "Null", "Structure", "Array" or "List. "value" will be mapped to an NSNumber, NSString, nil or NSArray
- * instance. When "type" is "OctetStriing", "value" will be an NSData object. When "type" is "Structure", "Array" or "List", "value"
- * will be NSArray with NSDictionary elements. Each NSDictionary element will have "tag" and "value" keys. "tag" will be mapped to
- * an NSNumber value. "value" will be mapped to an NSDictionary instance representing any attribute value recursively.
+ * @param completion  response handler will receive either values or error.
+ *
+ *                    Received values are an NSArray object with response-value element as described below.
+ *
+ *                    A response-value is an NSDictionary object with the following key values:
+ *
+ *                    kCHIPAttributePathKey : CHIPAttributePath object. Included for attribute value.
+ *                    kCHIPCommandPathKey : CHIPCommandPath object. Included for command response.
```

Comment:
It's a little odd to have this documented on readAttribute...  Maybe this whole big block actually be documentation for CHIPDeviceResponseHandler and the various functions should point to that documentation?